routes/demos.js

```diff
@@ -7,10 +7,94 @@ const fse = require('fs-extra');
 const config = require('config');
 const util = require('../libs/util');
 
+let projectsDeletionInProgress = false;
+
+function canCreateAnonymousProject(req, res, next) {
+  const anonymousProjectsLimit = 100;
+  Project.find({'user': req.user._id}, function (err, anonymousProjects) {
+    if (err) {
+      return res.status(400).send(err);
+    }
+
+    if (anonymousProjects.length >= anonymousProjectsLimit) {
+      tryDeleteOutDatedProjects(anonymousProjects);
+      return res.status(400).send('No possible to create more demos. Please try again later.');
+    }
+
+    next();
+  });
+};
+
+function tryDeleteOutDatedProjects(projects) {
+  if (projectsDeletionInProgress) {
+    console.log('Anonymous projects are being deleted ...');
+    return;
+  }
+
+  function isProjectOutDated(project) {
+    const projectCreatedDatePlusOneDay = new Date(project.created);
+    projectCreatedDatePlusOneDay.setDate(projectCreatedDatePlusOneDay.getDate() + 1 /* one day */);
+    return projectCreatedDatePlusOneDay <= Date.now();
+  }
+
+  const outDatedProjects = projects.filter(isProjectOutDated);
+  if (outDatedProjects.length === 0) {
+    return;
+  }
+
+  projectsDeletionInProgress = true;
+  let numberDeletedPojects = 0;
+
+  outDatedProjects.forEach(function(project) {
+    async.waterfall([
+      function (callback) {
+        // Remove the project using projectId
+        Project.remove({'_id': project._id}, function (error) {
+          if (error) {
+            return callback(error);
+          }
+
+          callback(null);
+        });
+      },
+      function (callback) {
+        // Remove the project folder
+        var projectPath = path.join(process.cwd(), 'projects', project._id.toString());
+        fse.remove(projectPath, function (err) {
+          if (err) {
+            return callback(err);
```

Comment:
if there is an error we return and don't try to delete project support folder,
I think we should try to delete what we can

Reply:
Ok I will revise this code.

Reply:
We treat project as deleted if it's successfully removed from db and project folder and its support folder are removed as well. This ensures that new project can be created.

Otherwise, if at least one condition failed, we report error since this is not obvious which one is critical or not to allow creating new project. And this project should be created manually.